File: Grafo/uri2720.py
class present:
    def __init__(self, id, x, y, z):
        self.id = id
        self.x = x
        self.y = y
        self.z = z


casos = int(input())
for i in range(casos):
    noel, bru = list(map(int, input().split()))
    lista = []
    for j in range(noel):
        id, x, y, z = list(map(int, input().split()))
        pre = present(id, x, y, z)
        lista.append(pre)
    lista.sort(key = lambda x: x.id)
    lista.sort(key = lambda x: x.x * x.y * x.z, reverse = True)
    
    listaBru = []
    for i in range(bru):
        listaBru.append(lista[i])
    listaBru.sort(key = lambda x: x.id)
    for i in listaBru:
        print(i.id, end=" ")
    print()

File: Grafo/uri2720.cpp
#include <bits/stdc++.h>

using namespace std;

struct Present{
	int id, x, y, z, tam;

	Present(int id, int x, int y, int z){
		this->id = id;
		this->x = x;
		this->y = y;
		this->z = z;
		tam = x * y * z;
	}
};

bool compareId(Present p1, Present p2){
	return (p1.id < p2.id);
}

bool comparePeso(Present p1, Present p2){
	return (p1.tam > p2.tam);
}

int main()
{
	int cases;
	scanf("%d", &cases);
	while(cases--){
		int noel, bru;
		scanf("%d %d", &noel, &bru);
		Present *p = (Present *) malloc(sizeof(Present) * noel);
		for(int i = 0; i < noel; i++){
			int id, x, y, z;
			scanf("%d %d %d %d", &id, &x, &y, &z);
			p[i] = Present(id, x, y, z);
		}

		sort(p, p + noel, compareId);
		stable_sort(p, p + noel, comparePeso);

		Present *bruno = (Present *) malloc(sizeof(Present) * bru);
		for(int i = 0; i < bru; i++){
			bruno[i] = Present(p[i].id, p[i].x, p[i].y, p[i].z);
		}
		sort(bruno, bruno + bru, compareId);
		bool espaco = true;
		for(int i = 0; i < bru; i++){
			if(espaco){
				printf("%d", bruno[i].id);
				espaco = false;
			}
			else{
				printf(" %d", bruno[i].id);
			}
		}
		printf("\n");
	}
}